src/test/java/io/managed/services/test/quickstarts/steps/JavaClientSteps.java

```diff
@@ -0,0 +1,141 @@
+package io.managed.services.test.quickstarts.steps;
+
+import io.managed.services.test.Environment;
+import io.managed.services.test.quickstarts.contexts.KafkaInstanceContext;
+import io.managed.services.test.quickstarts.contexts.ServiceAccountContext;
+import lombok.extern.log4j.Log4j2;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.JGitInternalException;
+import io.cucumber.java.en.When;
+import io.cucumber.java.en.Then;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertTrue;
+
+@Log4j2
+public class JavaClientSteps {
+
+    private File repository;
+
+    private static final String TMPDIR = "quickstart-repo";
+
+    private final KafkaInstanceContext kafkaInstanceContext;
+    private final ServiceAccountContext serviceAccountContext;
+
+    private Map<String, String> envsMap = new HashMap<>();
+
+    public JavaClientSteps(KafkaInstanceContext kafkaInstanceContext, ServiceAccountContext serviceAccountContext) {
+        this.kafkaInstanceContext = kafkaInstanceContext;
+        this.serviceAccountContext = serviceAccountContext;
+    }
+
+    // temporary usage to fetch examples from a different repository
+    @When("you clone the {word} repository from GitHub examples for Java")
```

Comment:
this is starting to be a bit repetitive (assuming it will be also imported in Python app quickstart). 
No need to refactor at the moment but maybe att smt. like "//TODO consider github context".